{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7e8040b2-da26-4065-9bee-f49ea52e7700",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting pdfplumber\n",
      "  Using cached pdfplumber-0.11.4-py3-none-any.whl.metadata (41 kB)\n",
      "Collecting python-docx\n",
      "  Using cached python_docx-1.1.2-py3-none-any.whl.metadata (2.0 kB)\n",
      "Collecting pdfminer.six==20231228 (from pdfplumber)\n",
      "  Using cached pdfminer.six-20231228-py3-none-any.whl.metadata (4.2 kB)\n",
      "Requirement already satisfied: Pillow>=9.1 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from pdfplumber) (10.3.0)\n",
      "Collecting pypdfium2>=4.18.0 (from pdfplumber)\n",
      "  Using cached pypdfium2-4.30.0-py3-none-win_amd64.whl.metadata (48 kB)\n",
      "Requirement already satisfied: charset-normalizer>=2.0.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from pdfminer.six==20231228->pdfplumber) (3.3.2)\n",
      "Requirement already satisfied: cryptography>=36.0.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from pdfminer.six==20231228->pdfplumber) (42.0.5)\n",
      "Requirement already satisfied: lxml>=3.1.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from python-docx) (5.2.1)\n",
      "Requirement already satisfied: typing-extensions>=4.9.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from python-docx) (4.11.0)\n",
      "Requirement already satisfied: cffi>=1.12 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from cryptography>=36.0.0->pdfminer.six==20231228->pdfplumber) (1.16.0)\n",
      "Requirement already satisfied: pycparser in c:\\users\\durge\\anaconda3\\lib\\site-packages (from cffi>=1.12->cryptography>=36.0.0->pdfminer.six==20231228->pdfplumber) (2.21)\n",
      "Using cached pdfplumber-0.11.4-py3-none-any.whl (59 kB)\n",
      "Using cached pdfminer.six-20231228-py3-none-any.whl (5.6 MB)\n",
      "Using cached python_docx-1.1.2-py3-none-any.whl (244 kB)\n",
      "Using cached pypdfium2-4.30.0-py3-none-win_amd64.whl (2.9 MB)\n",
      "Installing collected packages: python-docx, pypdfium2, pdfminer.six, pdfplumber\n",
      "Successfully installed pdfminer.six-20231228 pdfplumber-0.11.4 pypdfium2-4.30.0 python-docx-1.1.2\n"
     ]
    }
   ],
   "source": [
    "!pip install pdfplumber python-docx\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "db1e1c9f-19b7-4e5d-92df-80f7059ef4f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: unstructured[local-inference] in c:\\users\\durge\\anaconda3\\lib\\site-packages (0.15.13)\n",
      "Requirement already satisfied: chardet in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (4.0.0)\n",
      "Requirement already satisfied: filetype in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (1.2.0)\n",
      "Requirement already satisfied: python-magic in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (0.4.27)\n",
      "Requirement already satisfied: lxml in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (5.2.1)\n",
      "Requirement already satisfied: nltk in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (3.8.1)\n",
      "Requirement already satisfied: tabulate in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (0.9.0)\n",
      "Requirement already satisfied: requests in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (2.32.2)\n",
      "Requirement already satisfied: beautifulsoup4 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (4.12.3)\n",
      "Requirement already satisfied: emoji in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (2.13.2)\n",
      "Requirement already satisfied: dataclasses-json in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (0.6.7)\n",
      "Requirement already satisfied: python-iso639 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (2024.4.27)\n",
      "Requirement already satisfied: langdetect in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (1.0.9)\n",
      "Requirement already satisfied: numpy<2 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (1.26.4)\n",
      "Requirement already satisfied: rapidfuzz in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (3.10.0)\n",
      "Requirement already satisfied: backoff in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (2.2.1)\n",
      "Requirement already satisfied: typing-extensions in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (4.11.0)\n",
      "Requirement already satisfied: unstructured-client in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (0.25.9)\n",
      "Requirement already satisfied: wrapt in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (1.14.1)\n",
      "Requirement already satisfied: tqdm in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (4.66.4)\n",
      "Requirement already satisfied: psutil in c:\\users\\durge\\appdata\\roaming\\python\\python312\\site-packages (from unstructured[local-inference]) (6.0.0)\n",
      "Requirement already satisfied: python-oxmsg in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (0.0.1)\n",
      "Collecting unstructured.pytesseract>=0.3.12 (from unstructured[local-inference])\n",
      "  Using cached unstructured.pytesseract-0.3.13-py3-none-any.whl.metadata (11 kB)\n",
      "Requirement already satisfied: python-docx>=1.1.2 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (1.1.2)\n",
      "Collecting pikepdf (from unstructured[local-inference])\n",
      "  Using cached pikepdf-9.3.0-cp312-cp312-win_amd64.whl.metadata (8.4 kB)\n",
      "Collecting python-pptx>=1.0.1 (from unstructured[local-inference])\n",
      "  Using cached python_pptx-1.0.2-py3-none-any.whl.metadata (2.5 kB)\n",
      "Requirement already satisfied: pandas in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (2.2.2)\n",
      "Requirement already satisfied: markdown in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (3.4.1)\n",
      "Collecting google-cloud-vision (from unstructured[local-inference])\n",
      "  Using cached google_cloud_vision-3.7.4-py2.py3-none-any.whl.metadata (5.2 kB)\n",
      "Collecting unstructured-inference==0.7.36 (from unstructured[local-inference])\n",
      "  Using cached unstructured_inference-0.7.36-py3-none-any.whl.metadata (5.9 kB)\n",
      "Requirement already satisfied: openpyxl in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (3.1.2)\n",
      "Collecting pdf2image (from unstructured[local-inference])\n",
      "  Using cached pdf2image-1.17.0-py3-none-any.whl.metadata (6.2 kB)\n",
      "Collecting onnx (from unstructured[local-inference])\n",
      "  Using cached onnx-1.16.2-cp312-cp312-win_amd64.whl.metadata (16 kB)\n",
      "Collecting pypandoc (from unstructured[local-inference])\n",
      "  Using cached pypandoc-1.13-py3-none-any.whl.metadata (16 kB)\n",
      "Requirement already satisfied: pdfminer.six in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (20231228)\n",
      "Collecting xlrd (from unstructured[local-inference])\n",
      "  Using cached xlrd-2.0.1-py2.py3-none-any.whl.metadata (3.4 kB)\n",
      "Collecting pi-heif (from unstructured[local-inference])\n",
      "  Using cached pi_heif-0.18.0-cp312-cp312-win_amd64.whl.metadata (6.8 kB)\n",
      "Requirement already satisfied: networkx in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (3.2.1)\n",
      "Collecting effdet (from unstructured[local-inference])\n",
      "  Using cached effdet-0.4.1-py3-none-any.whl.metadata (33 kB)\n",
      "Requirement already satisfied: pypdf in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured[local-inference]) (5.0.1)\n",
      "Collecting layoutparser (from unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached layoutparser-0.3.4-py3-none-any.whl.metadata (7.7 kB)\n",
      "Collecting python-multipart (from unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached python_multipart-0.0.12-py3-none-any.whl.metadata (1.9 kB)\n",
      "Requirement already satisfied: huggingface-hub in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured-inference==0.7.36->unstructured[local-inference]) (0.24.7)\n",
      "Collecting opencv-python!=4.7.0.68 (from unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached opencv_python-4.10.0.84-cp37-abi3-win_amd64.whl.metadata (20 kB)\n",
      "Collecting onnxruntime>=1.17.0 (from unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached onnxruntime-1.19.2-cp312-cp312-win_amd64.whl.metadata (4.7 kB)\n",
      "Requirement already satisfied: matplotlib in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured-inference==0.7.36->unstructured[local-inference]) (3.8.4)\n",
      "Requirement already satisfied: torch in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured-inference==0.7.36->unstructured[local-inference]) (2.4.1)\n",
      "Collecting timm (from unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached timm-1.0.9-py3-none-any.whl.metadata (42 kB)\n",
      "Requirement already satisfied: transformers>=4.25.1 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured-inference==0.7.36->unstructured[local-inference]) (4.44.2)\n",
      "Requirement already satisfied: Pillow>=3.3.2 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from python-pptx>=1.0.1->unstructured[local-inference]) (10.3.0)\n",
      "Collecting XlsxWriter>=0.5.7 (from python-pptx>=1.0.1->unstructured[local-inference])\n",
      "  Using cached XlsxWriter-3.2.0-py3-none-any.whl.metadata (2.6 kB)\n",
      "Requirement already satisfied: packaging>=21.3 in c:\\users\\durge\\appdata\\roaming\\python\\python312\\site-packages (from unstructured.pytesseract>=0.3.12->unstructured[local-inference]) (24.1)\n",
      "Requirement already satisfied: soupsieve>1.2 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from beautifulsoup4->unstructured[local-inference]) (2.5)\n",
      "Requirement already satisfied: marshmallow<4.0.0,>=3.18.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from dataclasses-json->unstructured[local-inference]) (3.22.0)\n",
      "Requirement already satisfied: typing-inspect<1,>=0.4.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from dataclasses-json->unstructured[local-inference]) (0.9.0)\n",
      "Requirement already satisfied: torchvision in c:\\users\\durge\\anaconda3\\lib\\site-packages (from effdet->unstructured[local-inference]) (0.19.1)\n",
      "Collecting pycocotools>=2.0.2 (from effdet->unstructured[local-inference])\n",
      "  Using cached pycocotools-2.0.8-cp312-cp312-win_amd64.whl.metadata (1.1 kB)\n",
      "Collecting omegaconf>=2.0 (from effdet->unstructured[local-inference])\n",
      "  Using cached omegaconf-2.3.0-py3-none-any.whl.metadata (3.9 kB)\n",
      "Collecting google-api-core!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.1 (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.1->google-cloud-vision->unstructured[local-inference])\n",
      "  Using cached google_api_core-2.20.0-py3-none-any.whl.metadata (2.7 kB)\n",
      "Collecting google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1 (from google-cloud-vision->unstructured[local-inference])\n",
      "  Using cached google_auth-2.35.0-py2.py3-none-any.whl.metadata (4.7 kB)\n",
      "Collecting proto-plus<2.0.0dev,>=1.22.3 (from google-cloud-vision->unstructured[local-inference])\n",
      "  Using cached proto_plus-1.24.0-py3-none-any.whl.metadata (2.2 kB)\n",
      "Requirement already satisfied: protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<6.0.0dev,>=3.20.2 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from google-cloud-vision->unstructured[local-inference]) (3.20.3)\n",
      "Requirement already satisfied: six in c:\\users\\durge\\appdata\\roaming\\python\\python312\\site-packages (from langdetect->unstructured[local-inference]) (1.16.0)\n",
      "Requirement already satisfied: click in c:\\users\\durge\\anaconda3\\lib\\site-packages (from nltk->unstructured[local-inference]) (8.1.7)\n",
      "Requirement already satisfied: joblib in c:\\users\\durge\\anaconda3\\lib\\site-packages (from nltk->unstructured[local-inference]) (1.4.2)\n",
      "Requirement already satisfied: regex>=2021.8.3 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from nltk->unstructured[local-inference]) (2023.10.3)\n",
      "Requirement already satisfied: et-xmlfile in c:\\users\\durge\\anaconda3\\lib\\site-packages (from openpyxl->unstructured[local-inference]) (1.1.0)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in c:\\users\\durge\\appdata\\roaming\\python\\python312\\site-packages (from pandas->unstructured[local-inference]) (2.9.0.post0)\n",
      "Requirement already satisfied: pytz>=2020.1 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from pandas->unstructured[local-inference]) (2024.1)\n",
      "Requirement already satisfied: tzdata>=2022.7 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from pandas->unstructured[local-inference]) (2023.3)\n",
      "Requirement already satisfied: charset-normalizer>=2.0.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from pdfminer.six->unstructured[local-inference]) (3.3.2)\n",
      "Requirement already satisfied: cryptography>=36.0.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from pdfminer.six->unstructured[local-inference]) (42.0.5)\n",
      "Collecting Deprecated (from pikepdf->unstructured[local-inference])\n",
      "  Using cached Deprecated-1.2.14-py2.py3-none-any.whl.metadata (5.4 kB)\n",
      "Requirement already satisfied: olefile in c:\\users\\durge\\anaconda3\\lib\\site-packages (from python-oxmsg->unstructured[local-inference]) (0.47)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from requests->unstructured[local-inference]) (3.7)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from requests->unstructured[local-inference]) (2.2.2)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from requests->unstructured[local-inference]) (2024.8.30)\n",
      "Requirement already satisfied: colorama in c:\\users\\durge\\appdata\\roaming\\python\\python312\\site-packages (from tqdm->unstructured[local-inference]) (0.4.6)\n",
      "Requirement already satisfied: deepdiff>=6.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured-client->unstructured[local-inference]) (8.0.1)\n",
      "Requirement already satisfied: httpx>=0.27.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured-client->unstructured[local-inference]) (0.27.0)\n",
      "Requirement already satisfied: jsonpath-python>=1.0.6 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured-client->unstructured[local-inference]) (1.0.6)\n",
      "Requirement already satisfied: mypy-extensions>=1.0.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured-client->unstructured[local-inference]) (1.0.0)\n",
      "Requirement already satisfied: nest-asyncio>=1.6.0 in c:\\users\\durge\\appdata\\roaming\\python\\python312\\site-packages (from unstructured-client->unstructured[local-inference]) (1.6.0)\n",
      "Requirement already satisfied: requests-toolbelt>=1.0.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from unstructured-client->unstructured[local-inference]) (1.0.0)\n",
      "Requirement already satisfied: cffi>=1.12 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from cryptography>=36.0.0->pdfminer.six->unstructured[local-inference]) (1.16.0)\n",
      "Requirement already satisfied: orderly-set==5.2.2 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from deepdiff>=6.0->unstructured-client->unstructured[local-inference]) (5.2.2)\n",
      "Collecting googleapis-common-protos<2.0.dev0,>=1.56.2 (from google-api-core!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.1->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.1->google-cloud-vision->unstructured[local-inference])\n",
      "  Using cached googleapis_common_protos-1.65.0-py2.py3-none-any.whl.metadata (1.5 kB)\n",
      "Collecting grpcio<2.0dev,>=1.33.2 (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.1->google-cloud-vision->unstructured[local-inference])\n",
      "  Using cached grpcio-1.66.2-cp312-cp312-win_amd64.whl.metadata (4.0 kB)\n",
      "Collecting grpcio-status<2.0.dev0,>=1.33.2 (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.1->google-cloud-vision->unstructured[local-inference])\n",
      "  Using cached grpcio_status-1.66.2-py3-none-any.whl.metadata (1.1 kB)\n",
      "Requirement already satisfied: cachetools<6.0,>=2.0.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1->google-cloud-vision->unstructured[local-inference]) (5.3.3)\n",
      "Requirement already satisfied: pyasn1-modules>=0.2.1 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1->google-cloud-vision->unstructured[local-inference]) (0.2.8)\n",
      "Collecting rsa<5,>=3.1.4 (from google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1->google-cloud-vision->unstructured[local-inference])\n",
      "  Using cached rsa-4.9-py3-none-any.whl.metadata (4.2 kB)\n",
      "Requirement already satisfied: anyio in c:\\users\\durge\\anaconda3\\lib\\site-packages (from httpx>=0.27.0->unstructured-client->unstructured[local-inference]) (4.2.0)\n",
      "Requirement already satisfied: httpcore==1.* in c:\\users\\durge\\anaconda3\\lib\\site-packages (from httpx>=0.27.0->unstructured-client->unstructured[local-inference]) (1.0.2)\n",
      "Requirement already satisfied: sniffio in c:\\users\\durge\\anaconda3\\lib\\site-packages (from httpx>=0.27.0->unstructured-client->unstructured[local-inference]) (1.3.0)\n",
      "Requirement already satisfied: h11<0.15,>=0.13 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from httpcore==1.*->httpx>=0.27.0->unstructured-client->unstructured[local-inference]) (0.14.0)\n",
      "Collecting antlr4-python3-runtime==4.9.* (from omegaconf>=2.0->effdet->unstructured[local-inference])\n",
      "  Using cached antlr4_python3_runtime-4.9.3-py3-none-any.whl\n",
      "Requirement already satisfied: PyYAML>=5.1.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from omegaconf>=2.0->effdet->unstructured[local-inference]) (6.0.1)\n",
      "Collecting coloredlogs (from onnxruntime>=1.17.0->unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)\n",
      "Collecting flatbuffers (from onnxruntime>=1.17.0->unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached flatbuffers-24.3.25-py2.py3-none-any.whl.metadata (850 bytes)\n",
      "Requirement already satisfied: sympy in c:\\users\\durge\\anaconda3\\lib\\site-packages (from onnxruntime>=1.17.0->unstructured-inference==0.7.36->unstructured[local-inference]) (1.12)\n",
      "Requirement already satisfied: contourpy>=1.0.1 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from matplotlib->unstructured-inference==0.7.36->unstructured[local-inference]) (1.2.0)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from matplotlib->unstructured-inference==0.7.36->unstructured[local-inference]) (0.11.0)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from matplotlib->unstructured-inference==0.7.36->unstructured[local-inference]) (4.51.0)\n",
      "Requirement already satisfied: kiwisolver>=1.3.1 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from matplotlib->unstructured-inference==0.7.36->unstructured[local-inference]) (1.4.4)\n",
      "Requirement already satisfied: pyparsing>=2.3.1 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from matplotlib->unstructured-inference==0.7.36->unstructured[local-inference]) (3.0.9)\n",
      "Requirement already satisfied: safetensors in c:\\users\\durge\\anaconda3\\lib\\site-packages (from timm->unstructured-inference==0.7.36->unstructured[local-inference]) (0.4.5)\n",
      "Requirement already satisfied: filelock in c:\\users\\durge\\anaconda3\\lib\\site-packages (from torch->unstructured-inference==0.7.36->unstructured[local-inference]) (3.13.1)\n",
      "Requirement already satisfied: jinja2 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from torch->unstructured-inference==0.7.36->unstructured[local-inference]) (3.1.4)\n",
      "Requirement already satisfied: fsspec in c:\\users\\durge\\anaconda3\\lib\\site-packages (from torch->unstructured-inference==0.7.36->unstructured[local-inference]) (2024.3.1)\n",
      "Requirement already satisfied: setuptools in c:\\users\\durge\\anaconda3\\lib\\site-packages (from torch->unstructured-inference==0.7.36->unstructured[local-inference]) (69.5.1)\n",
      "Requirement already satisfied: tokenizers<0.20,>=0.19 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from transformers>=4.25.1->unstructured-inference==0.7.36->unstructured[local-inference]) (0.19.1)\n",
      "Requirement already satisfied: scipy in c:\\users\\durge\\anaconda3\\lib\\site-packages (from layoutparser->unstructured-inference==0.7.36->unstructured[local-inference]) (1.13.1)\n",
      "Collecting iopath (from layoutparser->unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached iopath-0.1.10-py3-none-any.whl\n",
      "Requirement already satisfied: pdfplumber in c:\\users\\durge\\anaconda3\\lib\\site-packages (from layoutparser->unstructured-inference==0.7.36->unstructured[local-inference]) (0.11.4)\n",
      "Requirement already satisfied: pycparser in c:\\users\\durge\\anaconda3\\lib\\site-packages (from cffi>=1.12->cryptography>=36.0.0->pdfminer.six->unstructured[local-inference]) (2.21)\n",
      "Collecting protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<6.0.0dev,>=3.20.2 (from google-cloud-vision->unstructured[local-inference])\n",
      "  Using cached protobuf-5.28.2-cp310-abi3-win_amd64.whl.metadata (592 bytes)\n",
      "Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from pyasn1-modules>=0.2.1->google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1->google-cloud-vision->unstructured[local-inference]) (0.4.8)\n",
      "Collecting humanfriendly>=9.1 (from coloredlogs->onnxruntime>=1.17.0->unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)\n",
      "Collecting portalocker (from iopath->layoutparser->unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached portalocker-2.10.1-py3-none-any.whl.metadata (8.5 kB)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from jinja2->torch->unstructured-inference==0.7.36->unstructured[local-inference]) (2.1.3)\n",
      "Requirement already satisfied: pypdfium2>=4.18.0 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from pdfplumber->layoutparser->unstructured-inference==0.7.36->unstructured[local-inference]) (4.30.0)\n",
      "Requirement already satisfied: mpmath>=0.19 in c:\\users\\durge\\anaconda3\\lib\\site-packages (from sympy->onnxruntime>=1.17.0->unstructured-inference==0.7.36->unstructured[local-inference]) (1.3.0)\n",
      "Collecting pyreadline3 (from humanfriendly>=9.1->coloredlogs->onnxruntime>=1.17.0->unstructured-inference==0.7.36->unstructured[local-inference])\n",
      "  Using cached pyreadline3-3.5.4-py3-none-any.whl.metadata (4.7 kB)\n",
      "Requirement already satisfied: pywin32>=226 in c:\\users\\durge\\appdata\\roaming\\python\\python312\\site-packages (from portalocker->iopath->layoutparser->unstructured-inference==0.7.36->unstructured[local-inference]) (306)\n",
      "Using cached unstructured_inference-0.7.36-py3-none-any.whl (56 kB)\n",
      "Using cached python_pptx-1.0.2-py3-none-any.whl (472 kB)\n",
      "Using cached unstructured.pytesseract-0.3.13-py3-none-any.whl (14 kB)\n",
      "Using cached effdet-0.4.1-py3-none-any.whl (112 kB)\n",
      "Using cached google_cloud_vision-3.7.4-py2.py3-none-any.whl (467 kB)\n",
      "Using cached onnx-1.16.2-cp312-cp312-win_amd64.whl (14.4 MB)\n",
      "Using cached pdf2image-1.17.0-py3-none-any.whl (11 kB)\n",
      "Using cached pi_heif-0.18.0-cp312-cp312-win_amd64.whl (1.7 MB)\n",
      "Using cached pikepdf-9.3.0-cp312-cp312-win_amd64.whl (3.5 MB)\n",
      "Using cached pypandoc-1.13-py3-none-any.whl (21 kB)\n",
      "Using cached xlrd-2.0.1-py2.py3-none-any.whl (96 kB)\n",
      "Using cached google_api_core-2.20.0-py3-none-any.whl (142 kB)\n",
      "Using cached google_auth-2.35.0-py2.py3-none-any.whl (208 kB)\n",
      "Using cached omegaconf-2.3.0-py3-none-any.whl (79 kB)\n",
      "Using cached onnxruntime-1.19.2-cp312-cp312-win_amd64.whl (11.1 MB)\n",
      "Using cached opencv_python-4.10.0.84-cp37-abi3-win_amd64.whl (38.8 MB)\n",
      "Using cached proto_plus-1.24.0-py3-none-any.whl (50 kB)\n",
      "Using cached pycocotools-2.0.8-cp312-cp312-win_amd64.whl (83 kB)\n",
      "Using cached timm-1.0.9-py3-none-any.whl (2.3 MB)\n",
      "Using cached XlsxWriter-3.2.0-py3-none-any.whl (159 kB)\n",
      "Using cached Deprecated-1.2.14-py2.py3-none-any.whl (9.6 kB)\n",
      "Using cached layoutparser-0.3.4-py3-none-any.whl (19.2 MB)\n",
      "Using cached python_multipart-0.0.12-py3-none-any.whl (23 kB)\n",
      "Using cached googleapis_common_protos-1.65.0-py2.py3-none-any.whl (220 kB)\n",
      "Using cached grpcio-1.66.2-cp312-cp312-win_amd64.whl (4.3 MB)\n",
      "Using cached grpcio_status-1.66.2-py3-none-any.whl (14 kB)\n",
      "Using cached protobuf-5.28.2-cp310-abi3-win_amd64.whl (431 kB)\n",
      "Using cached rsa-4.9-py3-none-any.whl (34 kB)\n",
      "Using cached coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n",
      "Using cached flatbuffers-24.3.25-py2.py3-none-any.whl (26 kB)\n",
      "Using cached humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
      "Using cached portalocker-2.10.1-py3-none-any.whl (18 kB)\n",
      "Using cached pyreadline3-3.5.4-py3-none-any.whl (83 kB)\n",
      "Installing collected packages: flatbuffers, antlr4-python3-runtime, XlsxWriter, xlrd, unstructured.pytesseract, rsa, python-multipart, pyreadline3, pypandoc, protobuf, portalocker, pi-heif, pdf2image, opencv-python, omegaconf, grpcio, Deprecated, python-pptx, proto-plus, pikepdf, onnx, iopath, humanfriendly, googleapis-common-protos, google-auth, pycocotools, grpcio-status, google-api-core, coloredlogs, timm, onnxruntime, layoutparser, google-cloud-vision, effdet, unstructured-inference\n",
      "  Attempting uninstall: protobuf\n",
      "    Found existing installation: protobuf 3.20.3\n",
      "    Uninstalling protobuf-3.20.3:\n",
      "      Successfully uninstalled protobuf-3.20.3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'C:\\\\Users\\\\durge\\\\anaconda3\\\\Lib\\\\site-packages\\\\cv2\\\\cv2.pyd'\n",
      "Consider using the `--user` option or check the permissions.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!pip install unstructured[local-inference]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4969b331-4c4a-44f5-ba94-ce65eb33fa46",
   "metadata": {},
   "outputs": [],
   "source": [
    "import nltk\n",
    "from nltk.tokenize import word_tokenize\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "462e8c74-a3ce-4c93-aa94-40a70d46c727",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package punkt to\n",
      "[nltk_data]     C:\\Users\\durge\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nltk.download('punkt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "082cc507-c429-40fc-8118-f9f5047233e0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
